cmd/spx/template/game.js

```diff
@@ -421,16 +422,67 @@ class GameApp {
         if (isClearIfDirty) {
             await this.updateProjectHash('')
             // clear the dirty cache
+            // TOOD only clear the current project's cache data
             this.clearPersistence(this.persistentPath);
             // create a default indexDB
-            await this.getObjectStore(this.persistentPath, 'FILE_DATA', 'readonly')
+            await this.ensureCacheDB()
         } else {
             await this.updateProjectHash(this.curProjectHash)
         }
         // cache is dirty, update it 
         await this.setCache(this.getProjectDataKey(), curData);
         return false;
     }
+    
+    async ensureCacheDB() {
+        await this.getObjectStore(this.persistentPath, 'FILE_DATA', 'readonly')
+    }
+
+    getEngineHashKey(assetName) {
+        return `${this.persistentPath}/.spx_engine_hash/${assetName}`
+    }
+    getEngineDataKey(assetName) {
+        return `${this.persistentPath}/.spx_engine_data/${assetName}`
+    }
+    async checkEngineCache() {
+        let hashes = GetEngineHashes()
+        this.logVerbose("curHashes ", hashes)
+        this.wasmGdspx = await this.checkEngineCacheAsset(hashes, "gdspx.wasm");
```

Comment:
I suggest that spx should not handle resource downloading cache by itself. When integrating & deploying within Builder, I will try to utilize the browser cache as much as possible to avoid users having to download the WASM file repeatedly. If spx do it again, on one hand, it will be more complicated if Builder implements optimizations like resource-preloading; on the other hand, there would also be more uncertainties for troubleshooting.

I would prefer if spx allowed me (the parent page) to specify a custom WASM URL, as that would make it easier for me to "utilize the browser's cache as much as possible." However, that is not an urgent issue.

Reply:
okay, I will remove this caching mechanism in a future PR and allow the WASM URL to be configurable. For now, we'll keep it as is

Reply:
@nighca 
This [commit](https://github.com/goplus/spx/pull/406/commits/8b87b9d0006987338e9b49cfa3f897422fe1baf8) Export engine asset URLs to config; disable asset cache mechanism by default 

You can specify the asset's URL with the argument `assetURLs`:
```js
window.startProject = async (buffer, projectName = "Game", showEditor = false, assetURLs = null)
```